frontend/src/Registration.jsx

```diff
@@ -0,0 +1,176 @@
+import {
+	Container,
+	TextField,
+	Grid,
+	Typography,
+	Button,
+} from "@material-ui/core";
+import { useForm } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as yup from "yup";
+
+const validationSchema = yup.object().shape({
+	firstName: yup.string().required(),
+	lastName: yup.string().required(),
+	email: yup.string().email().required(),
+	birthYear: yup.number().positive().required(),
+	birthMonth: yup.number().min(1).max(12).required(),
+	birthDay: yup.number().min(1).max(31).required(),
+	username: yup.string().required(),
+	password: yup.string().required().min(8),
+});
+
+function Registration() {
+	const { register, handleSubmit, errors: validationErrors } = useForm({
+		resolver: yupResolver(validationSchema),
+	});
+
+	function onSubmit(validatedData, event) {
+		event.preventDefault();
+		event.target.reset();
+
+		const formattedRegistrationData = {
+			name: {
+				first: validatedData.firstName,
+				last: validatedData.lastName,
+			},
+			email: validatedData.email,
+			dateOfBirth: {
+				year: validatedData.birthYear,
+				month: validatedData.birthMonth,
+				day: validatedData.birthDay,
+			},
+			username: validatedData.username,
+			password: validatedData.password,
+		};
+
+		console.log(formattedRegistrationData);
+	}
+
+	return (
+		<Container maxWidth="sm">
+			<Typography variant="h4">Register</Typography>
+			<form onSubmit={handleSubmit(onSubmit)}>
+				<Grid container spacing={2}>
+					<Grid item xs={12} sm={6}>
+						<TextField
+							label="First name"
+							name="firstName"
+							inputRef={register}
+							fullWidth
+							error={validationErrors.firstName ? true : false}
+							helperText={
```

Comment:
This is where we should take a minute to decide on an "internationalization approach" before we add a lot of components to the project. How can these messages be translated?